numpy/core/src/multiarray/dtype_transfer.c

```diff
@@ -2520,7 +2521,7 @@ _strided_to_strided_field_transfer(char *dst, npy_intp dst_stride,
 
 /*
  * Handles fields transfer.  To call this, at least one of the dtypes
- * must have fields
+ * must have fields. Does not take care of object<->structure conversion
```

Comment:
As I recall, this refers to converting object arrays (or python objects) to structures. 

If you go to the (only) place that `get_fields_transfer_function ` is called, note that it is in an if-block that excludes NPY_OBJECT types.